internal/page/donor/choose_attorneys.go

```diff
@@ -59,7 +60,9 @@ func ChooseAttorneys(tmpl template.Template, donorStore DonorStore, uuidString f
 				data.DobWarning = dobWarning
 			}
 
-			if data.Errors.Any() || data.Form.IgnoreNameWarning != nameWarning.String() {
+			if data.Errors.Any() ||
+				data.Form.IgnoreNameWarning != nameWarning.String() &&
+					attorney.FullName() != fmt.Sprintf("%s %s", data.Form.FirstNames, data.Form.LastName) {
```

Comment:
I feel like this would make more sense as `attorney.FirstNames != data.Form.FirstNames && attorney.LastName != data.Form.LastName` (and, not that I think people would ever do this, at the moment going from [A B] [C] to [A] [B C] would look like no change)

Reply:
also, the logic should be `nameHasntChanged && (anyErrors || warningNotIgnored)` otherwise if you:
1. make an attorney have the same name as the donor
2. accept the warning
3. go back to that attorney and make the dob invalid
4. you get the same name warning again